mos-platform/c64/lib/basic_header.S

```diff
@@ -1,11 +1,12 @@
-; C64 BASIC header that jumps to _start = 2061. This address immediately
+; C64 BASIC header that jumps to _start = 2062. This address immediately
 ; follows the header if it's at the beginning of BASIC memory ($0801).
 
 .section .basic_header,"aR",@progbits
 	.short next_line	; address of next BASIC line
 	.short 7773 		; line number: l33t for LLVM, a hint that this program was
 						; compiled with LLVM
 	.byte 0x9e			; SYS keyword BASIC token
-	.asciz "2061"		; the location of the _start symbol, in ASCII
+	.asciz "2062"		; the location of the _start symbol, in ASCII
 next_line:
 	.short 0   			; end of basic program
+	.byte 0             ; align to an odd number of bytes so that _start is at 2062 (2-byte aligned)
```

Comment:
No; at least not to my knowledge; the issue is that .text is already over-aligned to 2 bytes, generating the pad byte implicitly. For what it's worth I only see it when compiling Debug, without optimizations.

The basic header is already aligning at 1 byte, as required for it to start at byte 0x801.

Reply:
On this point-- I tried explicitly relaxing the alignment of the symbols that requested 2-byte alignement, but clang wouldn't do it.  It says minimum alignment is 2.

Reply:
Would you mind posting steps to reproduce the issue here? I smell a bug, and I'd at least like to give it a quick look before we work around it in this fashion.

Reply:
Sure; compile a program containing member functions, and instruct the linker to generate a map file.  You can see this in hello-new or hello-malloc.

When using the "Debug" cmake preset, the .text symbols in the map file will have 2-byte alignment.  The .text segment will specify 2 byte alignment.

When using "MinSizeRel", there are no .text symbols with 2-byte alignment.  The .text section has 1-byte alignment.

Reply:
Ah, I see the problem; because we hardcode the start address using ASCII, we can't allow the real start address of `.text` to vary with its alignment requirements. I'll make a separate bug for the issue; no action required here. Should be easy to fix, but some thought will be needed for the cleanest way.

Reply:
Created #14 for this issue.